### DISCUSSION TITLE
Unable to drop permissions from archived repositories makes archival date preservation tough

### DISCUSSION THREAD

**Discussion by jeffwilcox** [community/community#60868] [Repositories]
### Select Topic Area

Product Feedback

### Body

One of the challenges with the V3 REST API is that you cannot make permission changes to an archived repository. For example, trying to remove a collaborator from a repository that is archived yields the error:

```
  data: {
      message: 'Repository was archived so is read-only.',
      documentation_url: 'https://docs.github.com/rest/collaborators/collaborators#remove-a-repository-collaborator'
    }
  },
  request: {
    method: 'DELETE',
```

So if we are to cleanup former grants of permissions - which is especially important if someone was granted `admin` access to a repo as an Outside Collaborator, for example - we must first unarchive the repository, remove the collaborator or team grants, and then re-archive the repo.

The real challenge here is that the act of unarchiving and rearchiving, while technically it works, it then resets the archival date to "today"... vs when the repo was first archived... and I think we'd love to understand, for abandoned or old repos, the "true" "first" archival date instead.

Is there any technical reason for this limitation on removing permissions from an archived repository, or, any path to enabling this scenario - deleting access or teams from an archived repo - without first unarchiving?

**Comment by ahpook** [comment]
The limitation on removing permissions is just that it constitutes a change, which is generally disallowed to archived repos; we don't know or care (nor do I think we _should_) whether you're adding or removing permissions, just that you're changing something that's supposed to be closed down. 

Is the un- and re-archiving cycle something that you do pretty frequently? I guess I could see adding a "first_archived" date stamp that's only visible by the API, but it feels pretty niche.

**Reply by jeffwilcox** [reply]
I'm not sure it's that much of a change... I can add someone as an organization owner at any time, and they can then control and manage the archived repository.

Seems like a big security risk that essentially any archived repository will have permanently dangling collaborator grants potentially, requiring unarchival to make any permission changes. Typically, if I change the ACLs - teams or individuals - on a repo, it's not a "change", in there is no impact to the Git repository itself, there is no publicly visible change, the `updated_at` field is not changed for the repo, etc... 

Since a publicly archived repository shows the world the archival date, the unarchival action could potentially have a negative impact on the community's perception of how alive (or dead), or how recently active, a project was.

I agree the first archival date is too niche to deliver.

**Reply by lumaxis** [reply]
Maybe only allowing permission _removal_ could be a compromise here that makes sense? 🤔
I agree that having the archived date change is not ideal so I'd love to find a solution for this situation.

**Reply by jeffwilcox** [reply]
Maybe. This also breaks just-in-time solutions or other places where you might want to let someone temporarily get access to a private repo, even an archived one.

From a consistency perspective, I can still invite people to an archived organization.

There are cases where a private repo might have interesting information, or someone needs to evaluate whether to unarchive it, and requiring an existing person with access is not ideal to scalable access to it.

**Reply by guedesgomero** [reply]
Is this issue resolved?
